Command/CloneSiteCommand.php

```diff
@@ -0,0 +1,206 @@
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\PageBundle\Command;
+
+use Sonata\BlockBundle\Model\BlockInterface;
+use Sonata\PageBundle\Model\PageInterface;
+use Sonata\PageBundle\Model\SiteInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * Command for cloning complete sites including all pages and blocks.
+ *
+ * @author Armin Weihbold <armin.weihbold@gmail.com>
+ * @author Christian Gripp <mail@core23.de>
+ */
+final class CloneSiteCommand extends BaseCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function configure()
+    {
+        $this
+            ->setName('sonata:page:clone-site')
+            ->setDescription('Clone a complete site including all their pages')
+            ->addOption('sourceId', 's', InputOption::VALUE_REQUIRED, 'Source site id', null)
+            ->addOption('destId', 'd', InputOption::VALUE_REQUIRED, 'Destination site id', null)
+            ->addOption('prefix', 'p', InputOption::VALUE_REQUIRED, 'Title prefix', null)
+            ->addOption('onlyHybrid', 'oh', InputOption::VALUE_OPTIONAL, 'only clone hybrid pages', null);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function initialize(InputInterface $input, OutputInterface $output)
+    {
+        if (!$input->getOption('sourceId')) {
+            $this->listAllSites($output);
+
+            throw new \InvalidArgumentException('Please provide a "--sourceId=SITE_ID" option.');
+        }
+
+        if (!$input->getOption('destId')) {
+            $this->listAllSites($output);
+
+            throw new \InvalidArgumentException('Please provide a "--destId=SITE_ID" option.');
+        }
+
+        if (!$input->getOption('prefix')) {
+            throw new \InvalidArgumentException('Please provide a "--prefix=PREFIX" option.');
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($input->getOption('onlyHybrid')) {
+            $output->writeln('Cloning hybrid pages only.');
+            $hybridOnly = true;
+        } else {
+            $hybridOnly = false;
+        }
+
+        /** @var SiteInterface $sourceSite */
+        $sourceSite = $this->getSiteManager()->find($input->getOption('sourceId'));
+
+        /** @var SiteInterface $destSite */
+        $destSite = $this->getSiteManager()->find($input->getOption('destId'));
+
+        $pageClones = array();
+        $blockClones = array();
+
+        $output->writeln('Cloning pages');
+        /** @var PageInterface[] $pages */
+        $pages = $this->getPageManager()->findBy(array(
+            'site' => $sourceSite,
+        ));
```

Comment:
The cloning logic should be in a separate, unit-tested service, so that the clone can be done from something else than a command, like a rabbitmq consumer. The command should stay very skinny, just like a consumer or a controller.